File: 1122-relative-sort-array/1122-relative-sort-array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
var relativeSortArray = function(arr1, arr2) {
    let n = arr1.length;
    const arr = new Array(n);
    let idx = 0;
    arr1.sort((a, b) => a - b);
    for (let i = 0; i < arr2.length; i++) 
        for (let j = 0; j < n; j++) 
            if (arr1[j] === arr2[i]) {
                arr[idx++] = arr1[j];
                arr1[j] = -1;
            }
            
    for (let i = 0; i < n; i++) 
        if (arr1[i] !== -1) arr[idx++] = arr1[i];

    return arr;
};

File: 1122-relative-sort-array/1122-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int n = arr1.length;
        int[] arr = new int[n];
        int idx = 0;
        Arrays.sort(arr1);
        for (int i = 0; i < arr2.length; i++) {
            for (int j = 0; j < n; j++) {
                if (arr2[i] == arr1[j]) {
                    arr[idx++] = arr1[j];
                    arr1[j] = -1;
                }
            }
        }
        for (int i = 0; i < n; i++) 
            if (arr1[i] != -1) arr[idx++] = arr1[i];
            
        return arr;
    }
}